check_newest_file_age

```diff
@@ -168,9 +176,11 @@ while test -n "$1"; do
             ;;
         --check-dirs)
             check_dirs=1
+            # We want a recursive search (clear default value)
+            find_opts=''
```

Comment:
This implementation changes the behavior of how subdirectories are handled. In the current implementation, if directories are examined, they are only examined for the modification time of the directory itself, not any of the files and/or directories within it. This new implementation certainly has advantages, but also breaks backwards compatibility. I believe the correct way to implement this additional functionality is to a) model the existing behavior with find, and b) allow an additional argument to be passed that supports full recursive searches. For this pull request, I would accept implementing just a), or both a) and b), as long as they were implemented well :)